### ISSUE TITLE
`$ ./gradlew check` fails with Process 'command 'npm'' finished with non-zero exit value 1

### ISSUE THREAD

**Issue by takanamito** [treasure-data/digdag#753]
Hi.
I tried to run test according to [README](https:// github.com/treasure-data/digdag#running-tests)

and failed the task with
```
Execution failed for task ':digdag-ui:buildUi'.
> Process 'command 'npm'' finished with non-zero exit value 1
```

How to execute test?

- - - 

my environment

```
$ java -version                                                                                                                                                                                                                                                     
java version "1.8.0_162"
Java(TM) SE Runtime Environment (build 1.8.0_162-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.162-b12, mixed mode)

$ node -v
v7.10.1

$ npm -v
4.2.0
```

log

```
# /path/to/digdag
> $ ./gradlew check                                                                                                                                                                                                                                                    [master]

> Configure project :
The Task.leftShift(Closure) method has been deprecated and is scheduled to be removed in Gradle 5.0. Please use Task.doLast(Action) instead.
        at build_5ynz5r2501doid3nmjzg8scxz$_run_closure2.doCall(/Users/takanamito/.ghq/github.com/treasure-data/digdag/build.gradle:251)

Publication mavenJava not found in project :.
Publication mavenJava not found in project :digdag-docs.

> Task :digdag-ui:buildUi
npm WARN sass-loader@6.0.7 requires a peer of node-sass@^4.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN sass-loader@6.0.7 requires a peer of webpack@^2.0.0 || ^3.0.0 || ^4.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN digdag-ui@1.0.0 No repository field.
npm WARN digdag-ui@1.0.0 No license field.

up to date in 3.875s

> digdag-ui@1.0.0 build /Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui
> npm run clean && npm run build:js && npm run build:copy

npm WARN invalid config loglevel="notice"

> digdag-ui@1.0.0 clean /Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui
> rimraf public

npm WARN invalid config loglevel="notice"

> digdag-ui@1.0.0 build:js /Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui
> cross-env NODE_ENV=production webpack --config webpack.build.js -p --bail

=== Digdag ===
Config:  Release
Build path:  /Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/public
Output path:  assets/
Timestamp: 2018-04-09T05:55:13.696Z
Sha: assets
Error
    at NormalModule.onModuleBuildFailed (/Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/webpack-core/lib/NormalModuleMixin.js:315:19)
    at NormalModule.<anonymous> (/Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/webpack-core/lib/NormalModuleMixin.js:228:39)
    at runSyncOrAsync (/Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/webpack-core/lib/NormalModuleMixin.js:173:4)
    at NormalModule.loadPitch (/Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/webpack-core/lib/NormalModuleMixin.js:227:3)
    at NormalModule.doBuild (/Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/webpack-core/lib/NormalModuleMixin.js:241:4)
    at NormalModule.build (/Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/webpack/lib/NormalModule.js:84:14)
    at Compilation.buildModule (/Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/webpack/lib/Compilation.js:126:9)
    at /Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/webpack/lib/Compilation.js:309:10
    at /Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/webpack/lib/NormalModuleFactory.js:58:13
    at NormalModuleFactory.applyPluginsAsyncWaterfall (/Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/tapable/lib/Tapable.js:75:69)
    at onDoneResolving (/Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/webpack/lib/NormalModuleFactory.js:38:11)
    at onDoneResolving (/Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/webpack/lib/NormalModuleFactory.js:121:6)
    at /Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/webpack/lib/NormalModuleFactory.js:116:7
    at /Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/async/lib/async.js:726:13
    at /Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/async/lib/async.js:52:16
    at async.forEachOf.async.eachOf (/Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/async/lib/async.js:236:30)
    at _parallel (/Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/async/lib/async.js:717:9)
    at Object.async.parallel (/Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/async/lib/async.js:731:9)
    at /Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/webpack/lib/NormalModuleFactory.js:109:12
    at /Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/async/lib/async.js:726:13
    at /Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/async/lib/async.js:52:16
    at done (/Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/async/lib/async.js:246:17)
    at /Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/async/lib/async.js:44:16
    at /Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/async/lib/async.js:723:17
    at /Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/async/lib/async.js:167:37
    at /Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:29:4
    at onResolved (/Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/enhanced-resolve/lib/Resolver.js:39:10)
    at /Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/enhanced-resolve/lib/Resolver.js:123:21
    at /Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/enhanced-resolve/lib/Resolver.js:191:15
    at applyPluginsParallelBailResult.createInnerCallback.log (/Users/takanamito/.ghq/github.com/treasure-data/digdag/digdag-ui/node_modules/enhanced-resolve/lib/Resolver.js:104:30)

npm ERR! Darwin 17.5.0
npm ERR! argv "/Users/takanamito/.nodebrew/node/v7.10.1/bin/node" "/Users/takanamito/.nodebrew/current/bin/npm" "run" "build:js"
npm ERR! node v7.10.1
npm ERR! npm  v4.2.0
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! digdag-ui@1.0.0 build:js: `cross-env NODE_ENV=production webpack --config webpack.build.js -p --bail`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the digdag-ui@1.0.0 build:js script 'cross-env NODE_ENV=production webpack --config webpack.build.js -p --bail'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the digdag-ui package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     cross-env NODE_ENV=production webpack --config webpack.build.js -p --bail
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs digdag-ui
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls digdag-ui
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/takanamito/.npm/_logs/2018-04-09T05_55_18_118Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! digdag-ui@1.0.0 build: `npm run clean && npm run build:js && npm run build:copy`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the digdag-ui@1.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/takanamito/.npm/_logs/2018-04-09T05_55_18_134Z-debug.log


FAILURE: Build failed with an exception.

* Where:
Build file '/Users/takanamito/.ghq/github.com/treasure-data/digdag/build.gradle' line: 319

* What went wrong:
Execution failed for task ':digdag-ui:buildUi'.
> Process 'command 'npm'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED in 11s
48 actionable tasks: 1 executed, 47 up-to-date
```

**Comment by hiroyuki-sato** [comment]
Hello, @takanamito 

It seems that digdag needs update packages. 

I modified as follows. After that `digdag check` work correctly. 
But 'http://localhost:65432' doesn't show any contents.
It needs more fix.


```diff
diff --git a/digdag-ui/package.json b/digdag-ui/package.json
index c1b190ecb..2af09d411 100644
--- a/digdag-ui/package.json
+++ b/digdag-ui/package.json
@@ -69,14 +69,14 @@
     "less": "2.7.2",
     "less-loader": "2.2.3",
     "node-fs-extra": "^0.8.1",
-    "node-sass": "^3.10.0",
+    "node-sass": "^4.8.3",
     "resolve-url-loader": "2.0.2",
     "rimraf": "^2.5.4",
     "sass-loader": "^6.0.3",
     "standard": "^9.0.2",
     "style-loader": "0.16.1",
     "url-loader": "0.5.8",
-    "webpack": "1.13.1",
+    "webpack": "4.5.0",
     "webpack-dev-server": "1.14.1"
   },
   "standard": {
```

**Comment by takanamito** [comment]
@hiroyuki-sato Thank you. I completed to run `./gradlew check` with updating `package.json`.

> But 'http://localhost:65432' doesn't show any contents.
It needs more fix.

And should I keep this issue open?

**Comment by hiroyuki-sato** [comment]
Hello, @takanamito 

I created #755 issue. so please close this issue when you have a time.

**Comment by hiroyuki-sato** [comment]
@takanamito 
JFYI you don't need to apply [this patch](https://github.com/treasure-data/digdag/issues/753#issuecomment-379666013).
Instead, you need to apply #760. 
Or execute `npm install --save-dev resolve@1.6.0` in `digdag-ui` directory until mege #760.

Ref: [Twitter](https://twitter.com/i_szyn/status/984022265288421376) written in Japanese.

**Comment by hiroyuki-sato** [comment]
Hello, @takanamito 

Resolve 1.7.1 has been released. this version fixed resolve 1.7.0 issue.
So you don't need change package anymore.
Detail: https://github.com/treasure-data/digdag/pull/760#issuecomment-380722322

**Comment by takanamito** [comment]
Sorry for late response.
Thank you for fixing this issue. I succeed to run test.
